File: 2073-Time-Needed-to-Buy-Tickets.js
var timeRequiredToBuy = function(tickets, k) {
    let numUnitsTime = 0;
    let targetTickets = tickets[k];

    for (let i = 0; i < tickets.length; i++) {
        if (i <= k) {
            numUnitsTime += Math.min(targetTickets, tickets[i]);
        } else {
            numUnitsTime += Math.min(targetTickets - 1, tickets[i]);
        }
    }

    return numUnitsTime;
};

File: 2073-Time-Needed-to-Buy-Tickets.py
class Solution(object):
    def timeRequiredToBuy(self, tickets, k):
        numUnitsTime = 0

        for i, num in enumerate(tickets):
            if i <= k:
                numUnitsTime += min(tickets[k], tickets[i])
            else:
                numUnitsTime += min(tickets[k] - 1, tickets[i])

        return numUnitsTime
